### PR TITLE
ORC-1665: Migrate to `importOrder` of `spotless-maven-plugin`

### PR DESCRIPTION
**Pull request by cxzl25**
### What changes were proposed in this pull request?
This PR aims to enable the `importOrder` function in `spotless-maven-plugin`.

```bash
mvn spotless:apply -Panalyze
```

### Why are the changes needed?
Although there is now checkstyle to limit the order of imports, it cannot be automatically adjusted. `spotless-maven-plugin` can check and automatically adjust.

https://github.com/apache/orc/pull/1859#pullrequestreview-1959266086

### How was this patch tested?
local test

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: java/core/src/java/org/apache/orc/InMemoryKeystore.java [hunk 1]
```diff
@@ -330,4 +330,3 @@ private byte[] getMaterial() {
   }
 
 }
-
```
Comment (cxzl25) [inline java/core/src/java/org/apache/orc/InMemoryKeystore.java:333]: Because these files have two newlines at the end, one newline is automatically deleted after spotless apply.

File: java/bench/core/src/java/org/apache/orc/bench/core/RecordCounters.java [hunk 2]
```diff
@@ -49,4 +49,3 @@ public void print() {
     System.out.println("Invocations: " + invocations);
   }
 }
-
```
Comment (dongjoon-hyun) [inline java/bench/core/src/java/org/apache/orc/bench/core/RecordCounters.java:52]: Could you revert this change?

File: java/core/src/java/org/apache/orc/InMemoryKeystore.java (hunk 1 above)
Comment (dongjoon-hyun) [reply java/core/src/java/org/apache/orc/InMemoryKeystore.java:333]: > Because these files have two newlines at the end, one newline is automatically deleted after spotless apply.

Please make a new PR for the end line change separately before all of these migration.

Comment (dongjoon-hyun) [reply java/core/src/java/org/apache/orc/InMemoryKeystore.java:333]: Also, we need a way to prevent any future regression in terms of the end line.

Comment (cxzl25) [reply java/core/src/java/org/apache/orc/InMemoryKeystore.java:333]: > Please make a new PR for the end line change separately before all of these migration.

OK

> we need a way to prevent any future regression in terms of the end line

Now `spotless-maven-plugin` will check this behavior, so the beginning of the CI failed at the beginning.

https://github.com/apache/orc/actions/runs/8437582330/job/23107719304#step:5:8408

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by dongjoon-hyun** [comment]
Could you resolve the conflicts, @cxzl25 ?
